src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1811,7 +1811,12 @@ private function export($value)
 
     private function doExport($value, $resolveEnv = false)
     {
-        if (is_string($value) && false !== strpos($value, "\n")) {
+        if (is_string($value) && \in_array(preg_match('/[^\s\P{Cc}]/u', $value), array(false, 1), true)) {
```

Comment:
When can preg_match return `false` here?
What's the reasoning for choosing this regexp? (any reference/inspiration maybe?)

Reply:
If `$value` is not an UTF-8 character string, `preg_match` will return `false` because of the `u` modifier.
Regexp was chosen to detects all characters which are control characters `\p{Cc}` and not a space (which are valid ascii). I mean, putting a bell character in a string is very unlikely done in purpose. I don't know of a way to put this in a file except by copy/paste.

Reply:
Hum, I see. Then why unicode at all? No need for the `u` modifier, isn't it?

Reply:
If I not set `u` modifier, I can NOT use `\p{Cc}` to find the allowed characters as its extension for unicode. My first try was just selecting all characters outside standard ASCII but it would filter too much IMO.

Reply:
no need for unicode, let's list the ASCII Cc explicitly, here they are:
https://en.wikipedia.org/wiki/Unicode_control_characters

Reply:
Do you want a different regex for XML and PHP serialization? I mean, in XML, I need to check whether string contains only valid UTF-8 characters implying check with the `u` modifier.
For PHP, I can only check for `[\0-\x1f\x7f]` but then simple strings like `"Foo\tTab"` would be base64 encoded whereas `"\xf0"` would not.

Reply:
I'd suggest only one way to detect binary, and using this: `[\x00-\x08\x0B\x0E-\x1A\x1C-\x1F]`